activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -1159,8 +1168,8 @@ def add_pg_decoders
             "bool" => PG::TextDecoder::Boolean,
             "timestamp" => PG::TextDecoder::TimestampUtc,
             "timestamptz" => PG::TextDecoder::TimestampWithTimeZone,
-            "date" => PG::TextDecoder::Date,
           }
+          coders_by_name["date"] = PG::TextDecoder::Date if decode_dates
```

Comment:
Could be another PR, but I think it's worth deprecating this *not* being turned on. So in Rails 7.2 we introduce the new setting but it's opt-in, then in the next Rails version you can no longer opt out. What do you think?